### PR TITLE
Enhancement/search

### PR DESCRIPTION
**Pull request by iimpulse**
Fixing #61

### CODE REVIEW

File: src/main/java/org/jacksonlaboratory/repository/TermRepositoryImpl.java [hunk 1]
```diff
@@ -54,11 +54,16 @@ public List<OntologyTerm> findByTermIdIn(List<TermId> ids) {
...
 	@Transactional
 	public List<OntologyTerm> search(String searchTerm, boolean prefixSearch) {
+
+		if (searchTerm.trim().isEmpty()){
+			return Collections.emptyList();
+		}
+
 		if (prefixSearch){
-			return searchPartialId(searchTerm).collect(Collectors.toList());
+			return searchPartialId(searchTerm.trim()).collect(Collectors.toList());
```
Comment (gerring) [inline src/main/java/org/jacksonlaboratory/repository/TermRepositoryImpl.java:63]: Does it need null test and empty string test?

File: src/main/java/org/jacksonlaboratory/repository/TermRepositoryImpl.java [hunk 2]
```diff
@@ -54,11 +54,16 @@ public List<OntologyTerm> findByTermIdIn(List<TermId> ids) {
 	@Override
 	@Transactional
 	public List<OntologyTerm> search(String searchTerm, boolean prefixSearch) {
```
Comment (gerring) [inline src/main/java/org/jacksonlaboratory/repository/TermRepositoryImpl.java:56]: @NonNull on searchTerm maybe

**Review by ielis — COMMENTED** [review]
I'm not sure I know enough about the framework(s) to provide any insight here.
